server/athenian/api/internal/settings.py

```diff
@@ -727,29 +763,25 @@ async def list_release_matches(
                     ),
                 ),
             )
-        missing_logical = []
-        for repo in repos:
-            if repo not in loaded:
-                if RepositoryName.from_prefixed(repo).is_logical:
-                    missing_logical.append(repo)
-                    continue
-                settings.append(
-                    (
-                        repo,
-                        ReleaseMatchSetting(
-                            branches=default_branch_alias,
-                            tags=".*",
-                            events=".*",
-                            match=ReleaseMatch.tag_or_branch,
-                        ),
-                    ),
-                )
-        if missing_logical:
+        missing = [r for r in repo_names_to_collect if r not in loaded]
+        if missing_logical := [r for r in missing if r.is_logical]:
             raise ResponseError(
                 MissingSettingsError(
                     detail=f"Logical repositories must have release settings: {missing_logical}",
                 ),
             )
+        for missing_physical in [r for r in missing if not r.is_logical]:
```

Comment:
```suggestion
        missing_logical = []
        for r in repo_names_to_collect:
            if r in loaded:
                continue
            if r.is_logical:
                missing_logical.append(r)
                continue
            missing_physical = r
```
let's decrease the list comprehension bloat